app/src/main/java/com/youssef/fixit/Jobs/JobNetworkProvider.java

```diff
@@ -1,17 +1,22 @@
 package com.youssef.fixit.Jobs;
 
-import android.content.Context;
-
 import com.youssef.fixit.Models.Data.RetrofitClient;
+import com.youssef.fixit.Models.Error;
 import com.youssef.fixit.Models.Jobs.Jobs;
 
 import retrofit2.Call;
 import retrofit2.Callback;
 import retrofit2.Response;
 
 public class JobNetworkProvider implements IJobRepository {
+    Error error;
+
+    public JobNetworkProvider() {
+        error = new Error();
```

Comment:
Let's create the error object when we receive an error, like this every error is associated with the request, and also the life time of the error will be limited to the request failure instead